domain/src/main/java/org/oppia/android/domain/oppialogger/LoggingIdentifierController.kt

```diff
@@ -101,6 +110,14 @@ class LoggingIdentifierController @Inject constructor(
    */
   fun getSessionId(): DataProvider<String> = sessionIdDataProvider
 
+  /**
+   * Returns an in-memory data provider pointing to a class variable of [appSessionId].
+   *
+   * This ID is unique to each app session. A session starts when the app is opened and ends when
+   * the app is destroyed by the Android system.
+   */
+  fun getAppSessionId(): DataProvider<String> = appSessionIdDataProvider
```

Comment:
Please add tests for this & the flow below in the same way we verify other IDs in this controller.

Reply:
Done